packages/@aws-cdk/cdk/lib/context.ts

```diff
@@ -16,39 +18,56 @@ const SSM_PARAMETER_PROVIDER = 'ssm';
 export class ContextProvider {
 
   private readonly stack: Stack;
+  private readonly provider: string;
+  private readonly props: {[key: string]: any};
 
-  constructor(private context: Construct) {
+  constructor(private context: Construct, provider: string, props: {[key: string]: any} = {}) {
```

Comment:
If you declare the arguments as:

```
constructor(..., private readonly provider: string, private readonly props: {[key: string]: any} = {}) {
}
```

You don't have to declare the member variables above and don't have to assign them either, [that happens automatically](https://kendaleiv.com/typescript-constructor-assignment-public-and-private-keywords/).